tests/rdf-connect/ldes-client.test.ts

```diff
@@ -2989,4 +2981,927 @@ describe("Functional tests for the js:LdesClient RDF-Connect processor", () => {
         // Expect output stream to be closed
         expect(finished).toBeTruthy();
     });
+
+    test("Fetching an LDES unordered and checking if state is saved and enforced upon resume", async () => {
+        const outputStream1 = new SimpleStream<string>();
+
+        outputStream1.data((record) => {
+            // Check presence of SDS terms
+            expect(record.indexOf(SDS.stream)).toBeGreaterThan(0);
+            expect(record.indexOf(SDS.payload)).toBeGreaterThan(0);
+
+            // Extract timestamp property (ex:modified in this LDES)
+            const store = RdfStore.createDefault();
+            new Parser().parse(record).forEach((q) => store.addQuad(q));
+            // Check that all member data is present
+            expect(store.getQuads(null, EX.terms.subprop).length).toBeGreaterThan(0);
+        });
+
+        // Hack needed because for some reason this file remains after the test
+        // when using npm test. It does not happen when using bun though ¯\_(ツ)_/¯
+        if (fs.existsSync("./tests/data/save.json")) {
+            fs.rmSync(path.resolve("./tests/data/save.json"));
+        }
```

Comment:
Instead of having the following at the end of each test
```ts
 // Clean up
 fs.rmSync(path.resolve("./tests/data/save.json"));
```
we should add the following at line 84
```ts
    afterEach(() => {
        if (fs.existsSync("./tests/data/save.json")) {
            fs.rmSync(path.resolve("./tests/data/save.json"));
        }
    })
```
And then this hack is no longer needed, and the tests itself are also cleaner.

It might however even be better to mock the filesystem instead of actually writing to disk.

Reply:
I tried using `beforeEach` and it didn't work but it seems to work with `afterEach` for some reason. Adjusted in https://github.com/rdf-connect/ldes-client/pull/94/commits/18b6267c1fe2cb6593cd0e46a63b3a38f21eba62

Reply:
I now understand why this wasn't working. It turns out that the client writes down the state both when the notifier sends the [`close()` event](https://github.com/rdf-connect/ldes-client/blob/6e95279ecc204f7cf2d1dfa77ed9e2cb3500a75b/lib/client.ts#L406) and also when the [Node.js process is terminated](https://github.com/rdf-connect/ldes-client/blob/6e95279ecc204f7cf2d1dfa77ed9e2cb3500a75b/lib/client.ts#L233), which happens when the tests are done. That is why the state wasn't been cleared and the tests succeeded the first time but not after that. I made sure to delete the state before and after every test now https://github.com/rdf-connect/ldes-client/commit/dc102a2258e9df95d2e34e0fca9fb56f6e3995b8.